### PR TITLE
Added doPrivileged for setAccessible when parsing JPQL query with EXT…

### PR DESCRIPTION
**Pull request by ajaypaul-ibm**
Added doPrivileged for setAccessible when parsing JPQL query with EXTRACT for resolving issue:https://github.com/eclipse-ee4j/eclipselink/issues/1981

for https://github.com/eclipse-ee4j/eclipselink/issues/1981

### CODE REVIEW

**Comment by tkburroughs** [comment]
@lukasj Please review and advise on resolution to test failure. It does not appear that the test failure is related to the change, and we have not seen the failure locally.

**Review by lukasj — COMMENTED** [review]
same thing is there in `DefaultSemanticValidator` and `AbstractStateObject` - why not to fix it the same way there as well?

also do not forget to update copyright year in header, in this case from `2006, 2021` to `2006, 2023`

**Comment by ajaypaul-ibm** [comment]
@lukasj  I've made some changes to DefaultSemanticValidator and AbstractStateObject, and I've updated the copyright header. Could you please review them?

**Review by lukasj — APPROVED** [review]

**Comment by lukasj** [comment]
@ajaypaul-ibm this is required for master as well

**Comment by tkburroughs** [comment]
@lukasj Will the master branch continue to support running with a SecurityManager?  Java 21 has deprecated it, made it more difficult to enable, and Jakarta EE 11 has a minimum level of Java 21 and has removed the Java security requirement from all new/updated specifications. Or, since EclipseLink may be used outside of a Jakarata EE container environment, is the plan for EclipseLink to continue running with older Java levels and support the SecurityManager?

Basically, if EclipseLink 4.1/5.0 does not support the SecurityManager, then this wouldn't be needed in master.
